### PR TITLE
SUP-27927: fetch metadata from shared storage

### PR DESCRIPTION
**Pull request by amiras89**

### CODE REVIEW

File: plugins/metadata/lib/kMetadataKavaUtils.php [hunk 1]
```diff
@@ -89,8 +93,10 @@ protected static function getMetadataFieldValues($objectIds, $partnerId, $metada
 				}
 				else
 				{
-					$fullPath = realpath($fileSync['FILE_ROOT'] . $fileSync['FILE_PATH']);
+					$fullPath = kFile::realPath($fileSync['FILE_ROOT'] . $fileSync['FILE_PATH']);
```
Comment (yossipapi) [inline plugins/metadata/lib/kMetadataKavaUtils.php:96]: Better use kFile::getFileContent
Which supports both cases (file on NFS stoarge or file stored on S3).

File: plugins/metadata/lib/kMetadataKavaUtils.php [hunk 2]
```diff
@@ -17,6 +17,10 @@ class kMetadataKavaUtils
 	
 	protected static function getMetadataReadyFileSyncs($objectIds, $partnerId, $metadataProfileId)
 	{
+		$currentDcId = kDataCenterMgr::getCurrentDcId();
```
Comment (yossipapi) [inline plugins/metadata/lib/kMetadataKavaUtils.php:20]: I think its more nice to do:
$dcIds = kDataCenterMgr::getSharedStorageProfileIds();
$dcIds[] = kDataCenterMgr::getCurrentDcId();

Saves one line :)

File: plugins/metadata/lib/kMetadataKavaUtils.php (hunk 1 above)
Comment (amiras89) [reply plugins/metadata/lib/kMetadataKavaUtils.php:95]: Thanks for the tip!
Done (and tested)

File: plugins/metadata/lib/kMetadataKavaUtils.php (hunk 2 above)
Comment (amiras89) [reply plugins/metadata/lib/kMetadataKavaUtils.php]: agreed ;)

**Comment by yossipapi** [comment]
Reviewed

**Pull request merged**